### ISSUE TITLE
Numeric messages to SIS break messages

### ISSUE THREAD

**Issue by LordAro** [UniversityRadioYork/MyRadio#634]
Messages that are purely numeric in nature are being passed to sis.messages.update as a number type, which then [tries to call substring() on it](https://github.com/UniversityRadioYork/MyRadio/blob/master/src/Public/js/sis.messages.js#L91)

Not sure which is at fault here, the PHP that is sending the messages, or the JS for assuming strings (for both the title and the body fields), but something needs fixing

**Comment by LordAro** [comment]
Doesn't appear to be [the PHP](https://github.com/UniversityRadioYork/MyRadio/blob/master/src/Classes/ServiceAPI/MyRadio_Timeslot.php#L960)

```
[23:08 lordaro@ury ~] > php -a
Interactive mode enabled

php > var_dump(htmlspecialchars(1993));
string(4) "1993"
php > var_dump(htmlspecialchars("1993"));
string(4) "1993"
php > [23:09 lordaro@ury ~] > php -v
PHP 5.6.29 (cli) (built: Dec 24 2016 11:52:06) 
```

**Comment by AJDurant** [comment]
Could be in the json encoding of the messages

**Comment by LordAro** [comment]
I thought about that, but IIRC we had a similar issue a few years ago and fixed it at the source?

*searches*

https://github.com/UniversityRadioYork/MyRadio/issues/477 hmm, nope

regardless, htmlspecialchars always returns a string, so it's a string at the point that it leaves MyRadio_Timeslot::getMessages, so the same fix wouldn't apply here anyway

**Comment by AJDurant** [comment]
Yes but after that, I think when the data is serialised to json, if it
looks like a number it becomes a number.

**Comment by LordAro** [comment]
Confirmed: https://github.com/UniversityRadioYork/MyRadio/blob/master/src/Controllers/SIS/remote.php#L31 & https://github.com/UniversityRadioYork/MyRadio/blob/master/src/Classes/MyRadio/URLUtils.php#L62

**Comment by AJDurant** [comment]
We generally do want that, so just need a way to force messages to be
strings

**Comment by LordAro** [comment]
Probably shouldn't be closed just yet, although the original issue is now fixed
